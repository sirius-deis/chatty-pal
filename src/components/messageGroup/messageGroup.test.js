import { render, screen } from "@testing-library/react";
import MessageGroup from "./messageGroup";

const messages = [
  {
    id: 1,
    senderId: 1,
    message: "Message 1",
    type: "text",
    createdAt: new Date("2024-07-17T21:08:30").toISOString(),
  },
  {
    id: 2,
    senderId: 1,
    message: "Message 2",
    type: "text",
    createdAt: new Date("2024-07-17T21:09:50").toISOString(),
  },
];

const data = {
  messages,
  userId: 1,
}

describe("MessageGroup component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <MessageGroup {...data} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render messages", () => {
    render(<MessageGroup {...data} />);
    messages.forEach(({ message }) =>
      expect(screen.getByText(message)).toBeInTheDocument()
    );
  });
});
