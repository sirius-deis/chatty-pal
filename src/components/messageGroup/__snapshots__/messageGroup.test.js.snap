// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageGroup component should match snapshot 1`] = `
<div>
  <div
    class="sc-dAlyuH kJqaud"
    data-testid="message_group"
  >
    <div
      class="sc-kpDqfm jUJZSW  "
    >
      <p
        class="sc-gEvEer ePxOia"
      >
        Message 1
      </p>
    </div>
    <div
      class="sc-kpDqfm jUJZSW last "
    >
      <p
        class="sc-gEvEer ePxOia"
      >
        Message 2
      </p>
    </div>
  </div>
</div>
`;
